main.js

```diff
@@ -53,64 +55,42 @@ function generateRandomPalette() {
 	}
 }
 
-function updatePalette(array) {
-	for (var i = 0; i < array.length; i++) {
-		allBoxes[i].style.backgroundColor = array[i];
-		allHexes[i].innerText = array[i];
+function updatePalette(colorValues) {
+	for (var i = 0; i < colorValues.length; i++) {
+		allBoxes[i].style.backgroundColor = colorValues[i];
+		allHexes[i].innerText = colorValues[i];
 	}
 }
 
 function savePalette() {
-	var paletteCopy = currentPalette.slice();
-	savedPalettes.push(paletteCopy);
+	var savedPalette = currentPalette.slice();
```

Comment:
Changed name of the variable to better reflect the contents. currentPalette is essentially becoming savedPalette so each savedPalette is referenced instead of referencing the updated value of currentPalette.